[ATH-1114] - Create Account

meganduffy opened the pull request:
User can now register by providing their first name, last name, unique email and password.

jamesfairbairn-wpengine commented on Gemfile:
```diff
@@ -41,6 +44,8 @@ group :development do
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
+  gem 'shoulda-matchers'
+  gem 'rails-controller-testing'
```
I'm not sure, but I don't think you need to add this to both `:development, :test` and `test`. I think you could just add them to the `:test` one. Might need to check that though.

jamesfairbairn-wpengine commented on app/helpers/users_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
Unless you need this class it is empty, so you currently don't really need it.

jamesfairbairn-wpengine commented on app/models/user.rb:
```diff
@@ -0,0 +1,17 @@
+class EmailValidator < ActiveModel::EachValidator
```
You should really only have one class per file. You can move validators to `app/validators` see [here](https://stackoverflow.com/questions/42253003/where-should-i-put-the-custom-validators-in-rails-5)

meganduffy replied:
This is very useful, thank you!

jamesfairbairn-wpengine commented on spec/controllers/users_controller_spec.rb:
```diff
@@ -0,0 +1,50 @@
+
+RSpec.describe UsersController, type: :controller do
+
+  describe "GET #new" do
```
If you aren't going to write tests for the `new` endpoint, then it's probably a good idea to just remove this block for now.

jamesfairbairn-wpengine commented on spec/helpers/users_helper_spec.rb:
```diff
@@ -0,0 +1,15 @@
+# require 'rails_helper'
```
This file is completely commented out, so shouldn't really be included in source control

jamesfairbairn-wpengine commented on spec/controllers/users_controller_spec.rb:
```diff
@@ -0,0 +1,50 @@
+                                   email: 'unique@email.com', password: 'hard2guess' } } }
+      it "redirects to homepage" do
+        post :create, params: parameters
+        expect(response).to redirect_to "/"
```
I think you could replace `"/"` with `root_path`

jamesfairbairn-wpengine commented on spec/requests/users_request_spec.rb:
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Again, this file is basically empty, so we don't really need to include it.

jamesfairbairn-wpengine commented:
Few minor comments. Functionality-wise, it looks good. Also, did we decide to leave off using RuboCop for the moment? Just wondering, as if we didn't, there are a few violations at the moment.

meganduffy commented:
We are not using RuboCop currently, maybe it's something that could be revisited soon.

I implemented a few changes and removed those blank files, going to merge it in now!
